### PR TITLE
Add version 3

### LINKED ISSUE #26: Support series-3 repositories
In the scope of the activities (https://github.com/tarantool/tarantool/pull/8771, https://github.com/tarantool/tarantool/pull/8840, https://github.com/tarantool/tarantool/pull/8866, https://github.com/tarantool/rws/pull/79) on creating static build packages and repositories, we have to support the `series-3` repositories which significantly differ from `series-2` ones.

### PR DESCRIPTION
**Pull request by foxzi**
Add support for tarantool series-3.

Closes #26

### CODE REVIEW

File: installer.tpl.sh [hunk 1]
```diff
@@ -217,6 +217,13 @@ install_apt ()
   curl_check
   gpg_check
 
+  # For 3 series we use static builds and we can use one repository
+  # for all deb-based and rpm-based systems
+  if [ "${ver}" = 3 ]; then
+    os="linux-deb"
```
Comment (ylobankov) [inline installer.tpl.sh:223]: This will set such `os` and `dist` for modules as well, but it shouldn't. We have static build packages only for tarantool.

**Review by ylobankov — CHANGES_REQUESTED** [review]

File: installer.tpl.sh (hunk 1 above)
Comment (foxzi) [reply installer.tpl.sh]: I fixed it in another way.

File: installer.tpl.sh [hunk 2]
```diff
@@ -115,20 +115,20 @@ print_new_release_policy_and_exit ()
...
-    if [ "${ver}" = "2" ]; then
+    if [ "${ver}" = "2" ] || [ "${ver}" = "3" ]; then
       repo_path="pre-release/series-"
     else
       echo "'pre-release' repository can be set only with Tarantool series-2"
       print_new_release_policy_and_exit
     fi
   elif [ "${repo_type}" = "live" ]; then
-    if [ "${ver}" = "2" ]; then
+    if [ "${ver}" = "2" ] || [ "${ver}" = "3" ]; then
```
[the remark this answers is not in the document]
Comment (foxzi) [reply installer.tpl.sh:131]: fixed

File: installer.tpl.sh [hunk 3]
```diff
@@ -115,20 +115,20 @@ print_new_release_policy_and_exit ()
...
   if [ "${repo_type}" = "release" ]; then
-    if [ "${ver}" = "2" ]; then
+    if [ "${ver}" = "2" ] || [ "${ver}" = "3" ]; then
       repo_path="release/series-"
     else
       repo_path="release/"
     fi
   elif [ "${repo_type}" = "pre-release" ]; then
-    if [ "${ver}" = "2" ]; then
+    if [ "${ver}" = "2" ] || [ "${ver}" = "3" ]; then
```
[the remark this answers is not in the document]
Comment (foxzi) [reply installer.tpl.sh:124]: fixed

File: installer.tpl.sh [hunk 4]
```diff
@@ -235,8 +235,15 @@ install_apt ()
   curl -L "${gpg_key_url_modules}" | apt-key add -
 
   rm -f /etc/apt/sources.list.d/*tarantool*.list
-  echo "deb https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${repo_path}${ver}${gc64}/${os}/ ${dist} main" > ${apt_source_path}
-  echo "deb-src https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${repo_path}${ver}${gc64}/${os}/ ${dist} main" >> ${apt_source_path}
+  # For 3 series we use static builds and we can use one repository
+  # for all deb-based and rpm-based systems
+  if [ "${ver}" = 3 ]; then
```
Comment (ylobankov) [inline installer.tpl.sh:240]: It will be more readable if this condition is `if [ "${ver}" -le 2 ]`.

File: installer.tpl.sh [hunk 5]
```diff
@@ -235,8 +235,15 @@ install_apt ()
...
+    echo "deb https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${repo_path}${ver}${gc64}/linux-deb/ static main" > ${apt_source_path}
```
Comment (ylobankov) [inline installer.tpl.sh:241]: `${gc64}` is redundant for series-3.

File: installer.tpl.sh [hunk 6]
```diff
@@ -235,8 +235,15 @@ install_apt ()
...
+    echo "deb-src https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${repo_path}${ver}${gc64}/linux-deb/ static main" >> ${apt_source_path}
+  else
+    echo "deb https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${repo_path}${ver}${gc64}/${os}/ ${dist} main" > ${apt_source_path}
```
Comment (ylobankov) [inline installer.tpl.sh:244]: It is a copy-paste. It should be done in a more elegant way.

File: installer.tpl.sh [hunk 7]
```diff
@@ -267,7 +274,29 @@ install_yum_repo ()
     exit 1
   fi
 
-  cat <<EOF > /etc/yum.repos.d/tarantool_${ver_repo}.repo
+  # For 3 series we use static builds and we can use one repository
+  # for all deb-based and rpm-based systems
+  if [ "${ver}" = 3 ]; then
```
Comment (ylobankov) [inline installer.tpl.sh:279]: All comments above are applied to this as well.

**Review by ylobankov — CHANGES_REQUESTED** [review]

File: installer.tpl.sh [hunk 8]
```diff
@@ -115,20 +115,20 @@ print_new_release_policy_and_exit ()
 setup_type ()
```
[the remark this answers is not in the document]
Comment (foxzi) [reply installer.tpl.sh:120]: done

File: installer.tpl.sh (hunk 4 above)
Comment (foxzi) [reply installer.tpl.sh]: done

File: installer.tpl.sh (hunk 5 above)
Comment (foxzi) [reply installer.tpl.sh]: done

File: installer.tpl.sh (hunk 6 above)
Comment (foxzi) [reply installer.tpl.sh]: done

File: installer.tpl.sh (hunk 7 above)
Comment (foxzi) [reply installer.tpl.sh]: fixed

File: installer.tpl.sh [hunk 9]
```diff
@@ -272,24 +282,38 @@ install_yum_repo ()
     exit 1
   fi
 
+  # For 3 series we use static builds and we can use one repository
+  # for all deb-based and rpm-based systems
+  if [ "${ver}" -ge 3 ]; then
+    dist_url="${repo_path}${ver}"
+    os_dist="linux-rpm"
```
Comment (ylobankov) [inline installer.tpl.sh:289]: `os_dist` -> `dist_code`

File: installer.tpl.sh [hunk 10]
```diff
@@ -272,24 +282,38 @@ install_yum_repo ()
...
+    dist_code="static"
```
Comment (ylobankov) [inline installer.tpl.sh:290]: `dist_code` -> `dist_ver`

File: installer.tpl.sh [hunk 11]
```diff
@@ -272,24 +282,38 @@ install_yum_repo ()
...
+  else
+    dist_url="${repo_path}${ver}${gc64}"
+    os_dist="${dist}"
```
Comment (ylobankov) [inline installer.tpl.sh:293]: `os_dist` -> `dist_ver`

File: installer.tpl.sh [hunk 12]
```diff
@@ -272,24 +282,38 @@ install_yum_repo ()
...
-gpgkey=https://download.tarantool.org/tarantool/${repo_path}${ver}${gc64}/gpgkey
+name=${OS_NAME}-${dist_code} - Tarantool Sources
+baseurl=https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${dist_url}/${dist_code}/${os_dist}/SRPMS
+gpgkey=https://download.tarantool.org/tarantool/${dist_url}/gpgkey
 repo_gpgcheck=1
 gpgcheck=0
 priority=1
+EOF
 
+  cat <<EOF > /etc/yum.repos.d/tarantool_${ver_repo}.repo
```
Comment (ylobankov) [inline installer.tpl.sh:316]: Have you tested it at all???

File: installer.tpl.sh [hunk 13]
```diff
@@ -272,24 +282,38 @@ install_yum_repo ()
...
 repo_gpgcheck=1
 gpgcheck=0
 enabled=1
 priority=1
 
 [tarantool_${ver_repo}-source]
-name=${OS_NAME}-${dist} - Tarantool Sources
-baseurl=https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${repo_path}${ver}${gc64}/${OS_CODE}/${dist}/SRPMS
-gpgkey=https://download.tarantool.org/tarantool/${repo_path}${ver}${gc64}/gpgkey
+name=${OS_NAME}-${dist_code} - Tarantool Sources
```
Comment (ylobankov) [inline installer.tpl.sh:308]: Please return `dist` back.

**Review by ylobankov — CHANGES_REQUESTED** [review]

File: installer.tpl.sh [hunk 14]
```diff
@@ -120,21 +120,21 @@ print_new_release_policy_and_exit ()
...
     else
       repo_path="release/"
     fi
   elif [ "${repo_type}" = "pre-release" ]; then
-    if [ "${ver}" = "2" ]; then
+    if [ "${ver}" -ge "2" ]; then
       repo_path="pre-release/series-"
     else
-      echo "'pre-release' repository can be set only with Tarantool series-2"
+      echo "'pre-release' repository can be set only with Tarantool series-${ver}"
```
[the remark this answers is not in the document]
Comment (foxzi) [reply installer.tpl.sh]: fixed

File: installer.tpl.sh [hunk 15]
```diff
@@ -120,21 +120,21 @@ print_new_release_policy_and_exit ()
...
   if [ "${repo_type}" = "release" ]; then
-    if [ "${ver}" = "2" ]; then
+    if [ "${ver}" -ge "2" ]; then
       repo_path="release/series-"
     else
       repo_path="release/"
     fi
   elif [ "${repo_type}" = "pre-release" ]; then
-    if [ "${ver}" = "2" ]; then
+    if [ "${ver}" -ge "2" ]; then
```
[the remark this answers is not in the document]
Comment (foxzi) [reply installer.tpl.sh]: done

File: installer.tpl.sh [hunk 16]
```diff
@@ -120,21 +120,21 @@ print_new_release_policy_and_exit ()
 setup_type ()
 {
   if [ "${repo_type}" = "release" ]; then
-    if [ "${ver}" = "2" ]; then
+    if [ "${ver}" -ge "2" ]; then
```
[the remark this answers is not in the document]
Comment (foxzi) [reply installer.tpl.sh]: done

File: installer.tpl.sh [hunk 17]
```diff
@@ -120,21 +120,21 @@ print_new_release_policy_and_exit ()
...
       repo_path="pre-release/series-"
     else
-      echo "'pre-release' repository can be set only with Tarantool series-2"
+      echo "'pre-release' repository can be set only with Tarantool series-${ver}"
       print_new_release_policy_and_exit
     fi
   elif [ "${repo_type}" = "live" ]; then
-    if [ "${ver}" = "2" ]; then
-      echo "'live' repository can't be set with Tarantool series-2"
+    if [ "${ver}" -ge "2" ]; then
```
[the remark this answers is not in the document]
Comment (foxzi) [reply installer.tpl.sh]: done

File: installer.tpl.sh [hunk 18]
```diff
@@ -120,21 +120,21 @@ print_new_release_policy_and_exit ()
...
+      echo "'live' repository can't be set with Tarantool series-${ver}"
```
[the remark this answers is not in the document]
Comment (foxzi) [reply installer.tpl.sh]: OK

File: installer.tpl.sh (hunk 10 above)
Comment (foxzi) [reply installer.tpl.sh]: fixed

File: installer.tpl.sh (hunk 11 above)
Comment (foxzi) [reply installer.tpl.sh]: fixed

File: installer.tpl.sh (hunk 12 above)
Comment (foxzi) [reply installer.tpl.sh]: error fixed

File: installer.tpl.sh (hunk 13 above)
Comment (foxzi) [reply installer.tpl.sh]: fixed

**Comment by Totktonada** [comment]
It looks like that the interaction with the first reviewer is in progress. I'm going remove the request for review from me. Feel free to request it again when the review within your team will be finished.

File: installer.tpl.sh [hunk 19]
```diff
@@ -272,24 +282,38 @@ install_yum_repo ()
     exit 1
   fi
 
+  # For 3 series we use static builds and we can use one repository
+  # for all deb-based and rpm-based systems
+  if [ "${ver}" -ge 3 ]; then
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply installer.tpl.sh]: Ok, I fixed this.

File: installer.tpl.sh [hunk 20]
```diff
@@ -272,24 +282,38 @@ install_yum_repo ()
     exit 1
   fi
 
+  # For 3 series we use static builds and we can use one repository
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply installer.tpl.sh]: Ok, I fixed this.

File: installer.tpl.sh [hunk 21]
```diff
@@ -240,8 +240,18 @@ install_apt ()
   curl -L "${gpg_key_url_modules}" | apt-key add -
 
   rm -f /etc/apt/sources.list.d/*tarantool*.list
-  echo "deb https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${repo_path}${ver}${gc64}/${os}/ ${dist} main" > ${apt_source_path}
-  echo "deb-src https://download.tarantool.org{% if usr_id %}/{{ usr_id }}{% endif %}/tarantool/${repo_path}${ver}${gc64}/${os}/ ${dist} main" >> ${apt_source_path}
+  # For 3 series we use static builds and we can use one repository
+  # for all deb-based and rpm-based systems
+  if [ "${ver}" -ge 3 ]; then
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply installer.tpl.sh]: Ok, I fixed this.

**Review by ylobankov — APPROVED** [review]

**Review by Totktonada — APPROVED** [review]
I see no problems on a brief glance.

**Pull request merged**